agent/include/content_analysis/sdk/analysis_agent.h

```diff
@@ -47,62 +60,99 @@ namespace sdk {
 // can simply call Send().
 //
 // If the final verdict should be to block, the agent should first update the
-// response by calling SetSessionVerdictToBlock() before calling Send().
+// response by calling SetEventVerdictToBlock() before calling Send().
 //
-// This class is not thread safe.
+// This class is not thread safe.  However, it may be passed to another thread
+// as long as the agent properly serializses access to the event.
 //
 // See the demo directory for an example of how to use this class.
-class Session {
+class ContentAnalysisEvent {
  public:
-  virtual ~Session() = default;
+  virtual ~ContentAnalysisEvent() = default;
 
-  // Prepares the session for graceful shutdown.  Upon return calls to all
+  // Prepares the event for graceful shutdown.  Upon return calls to all
   // other methods of this class will fail.
   virtual int Close() = 0;
 
-  // Retrieves a reference to the content analysis request received from
-  // Google Chrome.
+  // Retrieves a read-only reference to the content analysis request received
+  // from Google Chrome.
   virtual const ContentAnalysisRequest& GetRequest() const = 0;
 
-  // Retrieves a reference to the content analysis response that will be
-  // sent to Google Chrome as the verdict for the request of this session.
-  // The agent may modify this response in place before calling Send.
+  // Retrieves a writable reference to the content analysis response that will
+  // be sent to Google Chrome as the verdict for the request of this event.
+  // The agent may modify this response in place before calling Send().
   virtual ContentAnalysisResponse& GetResponse() = 0;
 
-  // Retrieves a reference to the Acknowledgement sent by Google Chrome after
-  // the agent responds with a verdict.
-  virtual const Acknowledgement& GetAcknowledgement() const = 0;
-
   // Send the verdict to Google Chrome.  Once this method is called further
   // changes to the response are ignored.
   virtual int Send() = 0;
 
  protected:
-  Session() = default;
-  Session(const Session& rhs) = delete;
-  Session(Session&& rhs) = delete;
-  Session& operator=(const Session& rhs) = delete;
-  Session& operator=(Session&& rhs) = delete;
+  ContentAnalysisEvent() = default;
```

Comment:
Why are ctors/dtors deleted for this interface but not the AgentEventHandler one?
I think neither should have them, it's an interface so having the underlying type be copyable/movable shouldn't matter.

Reply:
Got it, then I'm ok with this.